### PR TITLE
[lint] add lintrunner support for circleci_linter

### PR DESCRIPTION
**Pull request by suo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #67942
* #67932
* #67936
* #67894
* #67890
* __->__ #67872

As title. This demonstrates some of the nice features of lintrunner:
- Uniform error reporting means you get a nice diff of the changes for
free
- Can run with -a to just accept the changes (don't need to tell people
to run a special regenerate command since the linter adaper already knows how.)

Differential Revision: [D32187386](https://our.internmc.facebook.com/intern/diff/D32187386)

### CODE REVIEW

**Comment by suo** [comment]
@suo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32187386).

**Comment by suo** [comment]
@suo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32187386).

**Comment by suo** [comment]
@suo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32187386).

**Comment by suo** [comment]
@suo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32187386).

File: tools/linter/adapters/circleci_linter.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+
+CHECKED_IN_FILE = "config.yml"
+REGENERATION_SCRIPT = "regenerate.sh"
+
+PARENT_DIR = os.path.basename(os.path.dirname(os.path.abspath(__file__)))
+README_PATH = os.path.join(PARENT_DIR, "README.md")
+
+
+class LintSeverity(str, Enum):
```
Comment (driazati) [inline tools/linter/adapters/circleci_linter.py:23]: Is there some reason we can't share these between all the lint adapters? One of my main things with the python lint runners right now is they've gotten pretty complicated over time (i.e. we have 1000+ lines to run clang-tidy) so if we do something new it should be understandable / extensible

**Review by driazati — APPROVED** [review]

**Pull request closed without merging**